core/arch/arm/include/mm/generic_ram_layout.h

```diff
@@ -0,0 +1,193 @@
+ * TEE_SDP_TEST_MEM_SIZE Set to CFG_TEE_SDP_MEM_SIZE or a default size.
+ *
+ * ----------------------------------------------------------------------------
+ * TEE RAM layout without CFG_WITH_PAGER
+ *_
+ *  +----------------------------------+ <-- CFG_TZDRAM_START
+ *  | TEE core secure RAM (TEE_RAM)    |
+ *  +----------------------------------+
+ *  | Trusted Application RAM (TA_RAM) |
+ *  +----------------------------------+
+ *  | SDP test memory (optional)       |
+ *  +----------------------------------+ <-- CFG_TZDRAM_START + CFG_TZDRAM_SIZE
+ *
+ *  +----------------------------------+ <-- CFG_SHMEM_START
+ *  | Non-secure static SHM            |
+ *  +----------------------------------+ <-- CFG_SHMEM_START + CFG_SHMEM_SIZE
+ *
+ * ----------------------------------------------------------------------------
+ * TEE RAM layout with CFG_WITH_PAGER=y and undefined CFG_TZSRAM_START/_SIZE
+ *
+ *  +----------------------------------+ <-- CFG_TZDRAM_START
+ *  | TEE core secure RAM (TEE_RAM)    |   | | CFG_CORE_TZSRAM_EMUL_SIZE
+ *  +----------------------------------+ --|-'
+ *  |   reserved (for kasan)           |   | TEE_RAM_VA_SIZE
+ *  +----------------------------------+ --'
+ *  | TA RAM / Pagestore (TA_RAM)      |
+ *  +----------------------------------+ <---- (TA_RAM is aligned on Mmu pgdir)
+ *  +----------------------------------+ <--
+ *  | SDP test memory (optional)       |   | CFG_TEE_SDP_MEM_SIZE
+ *  +----------------------------------+ <-+ CFG_TZDRAM_START + CFG_TZDRAM_SIZE
+ *
+ *  +----------------------------------+ <-- CFG_SHMEM_START
+ *  | Non-secure static SHM            |   |
+ *  +----------------------------------+   v CFG_SHMEM_SIZE
+ *
+ * ----------------------------------------------------------------------------
+ * TEE RAM layout with CFG_WITH_PAGER=y and define CFG_TZSRAM_START/_SIZE
+ *
+ *  +----------------------------------+ <-- CFG_TZSRAM_START
+ *  | TEE core secure RAM (TEE_RAM)    |   | CFG_TZSRAM_SIZE
+ *  +----------------------------------+ --'
+ *
+ *  +----------------------------------+  <- CFG_TZDRAM_START
+ *  | TA RAM / Pagestore (TA_RAM)      |
+ *  |----------------------------------+ <---- (TA_RAM is aligned on Mmu pgdir)
+ *  |----------------------------------+ <--
+ *  | SDP test memory (optional)       |   | CFG_TEE_SDP_MEM_SIZE
+ *  +----------------------------------+ <-+ CFG_TZDRAM_START + CFG_TZDRAM_SIZE
+ *
+ *  +----------------------------------+ <-- CFG_SHMEM_START
+ *  | Non-secure static SHM            |   |
+ *  +----------------------------------+   v CFG_SHMEM_SIZE
+ */
+
+#ifdef CFG_TEE_LOAD_ADDR
+#define TEE_LOAD_ADDR		CFG_TEE_LOAD_ADDR
+#else
+/* Platform specific platform_config.h may set TEE_LOAD_ADDR */
+#endif
+
+#ifdef CFG_TEE_VA_SIZE
+#define TEE_VA_SIZE		CFG_TEE_VA_SIZE
+#else
+#define TEE_VA_SIZE		CORE_MMU_PGDIR_SIZE
+#endif
+
+#ifdef CFG_SHMEM_SIZE
+#define TEE_SHMEM_SIZE		CFG_SHMEM_SIZE
+#else
+#define TEE_SHMEM_SIZE		SIZE_2M
```

Comment:
The static SHM range has to be kept in sync with the FDT passed to the kernel so the platform can't really rely on some default value.

Reply:
This is true for the address and size of the SHM, and also the secure DDR. All platform do not access the Linux kernel FDT. So, it is up to the platform to ensure linux config (static or fdt) is consistent with optee_os configuration when modifying the platform default config.

Well, yet I agree the `#else` part should better be removed.